### ISSUE TITLE
Build fails with gcc-13

### ISSUE THREAD

**Issue by tbe** [microsoft/DirectXShaderCompiler#5238]
With GCC-13, the build fails with:

```
/var/tmp/portage/dev-util/DirectXShaderCompiler-1.7.2212/work/DirectXShaderCompiler-1.7.2212/include/llvm/Analysis/DxilValueCache.h:55:16: error: declaration of ‘llvm::DxilValueCache::WeakValueMap llvm::DxilValueCache::ValueMap’ changes meaning of ‘ValueMap’ [-Wchanges-meaning]
   55 |   WeakValueMap ValueMap;
      |                ^~~~~~~~
/var/tmp/portage/dev-util/DirectXShaderCompiler-1.7.2212/work/DirectXShaderCompiler-1.7.2212/include/llvm/Analysis/DxilValueCache.h:40:5: note: used here to mean ‘class llvm::ValueMap<const llvm::Value*, llvm::DxilValueCache::WeakValueMap::ValueEntry>’
   40 |     ValueMap<const Value *, ValueEntry> Map;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /var/tmp/portage/dev-util/DirectXShaderCompiler-1.7.2212/work/DirectXShaderCompiler-1.7.2212/include/llvm/Analysis/DxilValueCache.h:14:
/var/tmp/portage/dev-util/DirectXShaderCompiler-1.7.2212/work/DirectXShaderCompiler-1.7.2212/include/llvm/IR/ValueMap.h:80:7: note: declared here
   80 | class ValueMap {
      |       ^~~~~~~~
```

**Comment by SteveEbey73742** [comment]
is it a bad code issue, or bug in GCC?

**Comment by llvm-beanz** [comment]
Fixed in #5179.
